bootstrap.sh

```diff
@@ -468,13 +468,6 @@ build_cmake_command(){
 
   add_os_flags
 
-  curl -V | grep OpenSSL &> /dev/null
```

Comment:
This was a fundamentally flawed way to try to determine the SSL backend used by curl and make decisions based on this information for the following reasons:
* What SSL backend the command line curl tool on the build machine is built with, bears practically no relevance to what the curl library that ends up invoked in the code is built with.
  * First, while unlikely, the command line tool on the system might not use the libcurl on the system that we will link with.
  * Second, it is not at all guaranteed that we will link with the system libcurl - we might build our own with USE_SYSTEM_CURL=OFF.
  * Thirdly, even if we do dynamically link the system libcurl, the libcurl on the target system, where the depencency will be loaded might be built with an another SSL backend.

Therefore, if we want to determine what SSL backend the libcurl we end up using has, it must be done at runtime.
While this would be possible - even if not elegant - to do by invoking `curl_version_info`, I propose that we should not even try to do this at runtime.

The only purpose we used USE_CURL_NSS for, was to determine whether we should set the SSL information through curl settings or do it directly by interacting with the SSL_CTX.
Even if we somehow determined that our curl used OpenSSL, so that we can use CURLOPT_SSL_CTX_FUNCTION, it is not a good idea to do it.
It is not at all guaranteed that the OpenSSL we are linked with is the same one curl is built with. And if that's the case, then curl will create the SSL_CTX with its own OpenSSL version that we will try to manipulate with an another OpenSSL version, possibly crashing the whole thing.

The only configuration this could work with, is if we are linked with the system libcurl, system OpenSSL, and the system libcurl is linked with the same system OpenSSL. While this is a common configuration, it only represents one configuration out of the many we can be built with.

Additionally, because I see no benefit (in functionality or ease of use), or certainly not one that would outweigh the risks, to use the SSLContextService's `configure_ssl_context` to set up curl's SSL instead of extracting the information from the SSLContextService and passing it properly to curl, and there are many ways in which we can fail if we try to do it, I have removed that path from the code.

Reply:
If we build both libcurl and libressl ourselves, then the risks I outlined would be mitigated, because we can be sure of the SSL backend libcurl uses and we can be sure that we use the exact same SSL library curl does, so we will be able to use configure_ssl_context.
However, by doing this we can not avoid changes entirely.
This current method of determining whether we should use configure_ssl_context (USE_CURL_NSS) would still have to be changed - if we limit ourselves to a statically linked libressl and libcurl, then the answer is we should always use that and we don't really need the curl setopt way of doing it.
If the main risk of change we want to avoid is switching to the curl setopt method of configuring SSL where formerly a configure_ssl_context has been used, then yes, doing this would mitigate that risk of change as well.
While I think this would be a good solution for now, in the long term I would actually like us to use the curl setopt method - it is more versatile and does not limit us to this configuration. But I do understand that to mitigate the risk of doing that we would have to do extensive testing.